File: input.hs
module Main where

import System.IO

-- Función para obtener las letras de un nombre
obtenerLetras :: String -> [Char]
obtenerLetras = filter (/= ' ')

-- Función principal
main :: IO ()
main = do
    putStr "Por favor, ingresa tu nombre: "
    hFlush stdout
    nombreUsuario <- getLine
    
    if null nombreUsuario
        then putStrLn "No ingresaste ningún nombre. ¡Hasta luego!"
        else do
            putStrLn $ "\n¡Hola, " ++ nombreUsuario ++ "!"
            putStrLn "Las letras de tu nombre son:"
            mapM_ (\c -> putStrLn $ "- " ++ [c]) (obtenerLetras nombreUsuario)
            
            putStr "¿Cuántos años tienes?: "
            hFlush stdout
            edadStr <- getLine
            let edad = read edadStr :: Int
                minEdad = 18
            
            if edad >= minEdad
                then putStrLn "¡Eres mayor de edad!"
                else putStrLn $ "Te faltan " ++ show (minEdad - edad) ++ " años para ser mayor de edad."
            
            putStrLn "\n¡Programa terminado!"

File: input.py
def main():

    nombre_usuario = input("Por favor, ingresa tu nombre: ")

    if not nombre_usuario:
        print("No ingresaste ningún nombre. ¡Hasta luego!")
        return 

    print(f"\n¡Hola, {nombre_usuario}!") 

    print("Las letras de tu nombre son:")

    letras_nombre = [] 
    for letra in nombre_usuario:
        if letra != ' ':
            letras_nombre.append(letra) 
    if letras_nombre:
        for l in letras_nombre:
            print(f"- {l}")

    edad = input("¿Cuántos años tienes?: ")
    min_edad = 18

    if int(edad) >= min_edad:
        print("¡Eres mayor de edad!")
    else:
        diff = min_edad - int(edad)
        print("Te faltan", diff, "años para ser mayor de edad.")

    
    print("\n¡Programa terminado!")


main() 